proposals/public-suffix.md

```diff
@@ -0,0 +1,227 @@
+
+### Abuse Mitigations
+
+This does not expose any new non-public data so there are no new abuse vectors.
+
+### Additional Security Considerations
+
+N/A
+
+## Alternatives
+
+### Existing Workarounds
+
+Developers can download the PSL dataset, bundle it with their extensions, and implement
+logic that parses and interprets the dataset in order to determine the topmost
+*registrable domain / eTLD+1* for a domain name. There are several drawbacks to
+this approach:
+
+1. Potential for inconsistencies in the determination of topmost *registrable domain / eTLD+1*
+by the extension and the host browser, due to differences in the version
+of the PSL dataset used, and differences in the implementations that the host
+browser and the extension use in order to interpret this dataset.
+2. Increased complexity and maintenance costs of extensions.
+3. Increased performance overhead of extensions due to bundling of bulky
+PSL dataset, leading to increase in memory, disk usage and increase in CPU usage
+due to possibly suboptimal extension implementations and repeating work already
+done by the host browser.
+
+### Open Web API
+
+The purpose of this API is to eliminate the potential for inconsistency between
+the host browser and its hosted extensions. The simplest way of achieving this
+is for extensions to access this functionality via the host browser itself rather
+than via some external source, such as an Open Web API.
+
+It is then a determination for the host browser itself as to whether
+the functionality (used by both the host browser and its extensions)
+should ultimately be obtained by means of an Open Web API.
+
+## Implementation Notes
+
+Since the major browsers all already implement internal methods for determining
+topmost *registrable domain / eTLD+1*s, it is hoped that the implementation will
+involve little more than providing the relevant mechanism for exposing these same methods
+to extensions:
+
+| Browser | Registrable Domain Method |
+| ------- | ------------------------- |
+| Chrome  | [GetDomainAndRegistry](https://source.chromium.org/chromium/chromium/src/+/main:net/base/registry_controlled_domains/registry_controlled_domain.h;l=182;drc=4f516be19f2c1d35dc3240d050d84d10f0d6f726) |
+| Firefox | [getBaseDomain](https://searchfox.org/mozilla-central/rev/e9f9bf31d1c0057a1cd339b5a853a75d1b16db39/netwerk/dns/nsIEffectiveTLDService.idl#94) |
+| Safari  | [topPrivatelyControlledDomain](https://github.com/WebKit/WebKit/blob/01eba7c416725cfd4eec57ab16daffa25b8124b4/Source/WebCore/platform/PublicSuffixStore.h#L43) |
+
+## Future Work
+
+### 1. Extend the API
+
+The major browsers provide additional methods/parameters internally for getting
+information related to the *registable domain / eTLD+1*. The API could be extended
+to expose more of these internal methods/parameters, for example:
+
+1. Provide an option to include private domains when getting the *registrable domain / eTLD+1*.
```

Comment:
Could you say more about why this is considered future work? I think it would be quite important to have, and even wonder if it should be enabled by default. Are there common use cases where you wouldn't want to include the private section?

Reply:
I agree it may be better to have the private section included by default as you suggest, and am happy to make this change. Here are some points to consider:

#### 1. Transparency / simplicity
By excluding the private section, the API returns only domains registered with an ICANN registrar. With the private section included, the registry mechanism for the returned domain may be less clear.

##### Example

| Type | Domain |
| - | - |
| Full Domain to lookup | foo.bar.baz.execute-api.us-east-1.amazonaws.com |
| Registrable Domain (excl. private) | amazonaws.com |
| Registrable Domain (incl. private) | execute-api.us-east-1.amazonaws.com |


#### 2. Granularity
This API may be used to automatically divide up lists of URLs into groups by their *registrable domain*. If using the entries from the private section, you are likely to end up with a higher number of groups, and fewer URLs per group (i.e. more fine-grained grouping). For example:

##### Example: URLs to be Sorted Into Groups

| URLs |
| - |
| foo.execute-api.us-east-1.amazonaws.com/ |
| bar.emrstudio-prod.us-east-1.amazonaws.com/ |
| baz.s3.dualstack.us-east-1.amazonaws.com/ |

##### No Private Domains

| Registrable Domain (excl. private) | URLs |
| - | - |
| **amazonaws.com** | |
|   | foo.execute-api.us-east-1.amazonaws.com/       |
|   | bar.emrstudio-prod.us-east-1.amazonaws.com/ |
|   | baz.s3.dualstack.us-east-1.amazonaws.com/      |

##### Including Private Domains

| Registrable Domain (incl. private) | URLs |
| - | - |
| **execute-api.us-east-1.amazonaws.com**    |  |
|  | foo.execute-api.us-east-1.amazonaws.com/ |
| **emrstudio-prod.us-east-1.amazonaws.com** |  |
|   | bar.emrstudio-prod.us-east-1.amazonaws.com/ |
| **s3.dualstack.us-east-1.amazonaws.com**   |  |
|   | baz.s3.dualstack.us-east-1.amazonaws.com/ |

Reply:
Done - now defaults to Private suffixes included 638833b2044492ca4b99415c1e5762e3c969d3e1